server/src/main/kotlin/org/octopusden/octopus/dms/service/impl/AdminServiceImpl.kt

```diff
@@ -68,6 +74,80 @@ class AdminServiceImpl( //TODO: move functionality to ComponentService and Artif
         }
     }
 
+    /**
+     * Update component name and return new component name.
+     * By this point, the component should have already been renamed in 'releng'.
+     * @param name - old component name
+     * @param newName - new component name
+     * @param dryRun - if true, do not update component name
+     * @throws NotFoundException if component with name [name] not found in releng
+     */
+    @Transactional(readOnly = false)
+    override fun renameComponent(name: String, newName: String, dryRun: Boolean) {
+        log.info("Update component name from '$name' to '$newName'")
+
+        if (isComponentPresentInRegistry(name)) {
+            log.error("Component with name $name exists in components registry")
+            throw IllegalComponentRenamingException("Component with name $name exists in components registry")
+        }
+        if (!isComponentPresentInRegistry(newName)) {
+            log.error("Component with name $newName not found in components registry")
+            throw NotFoundException("Component with name $newName not found in components registry")
+        }
+        checkComponentExistsInReleng(newName)
+        val component = componentRepository.findByName(newName)
+        val existedComponent = componentRepository.findByName(name)
+
+        if (existedComponent != null && component != null) {
+            log.error("Component with name $name and component with name $newName exists in DMS")
+            throw IllegalComponentRenamingException("Component with name $newName already exists in DMS")
+        }
+
+        existedComponent?.let {
+            if (!dryRun) {
+                componentRepository.save(Component(name = newName, id = it.id))
+                log.info("Component with name $name updated to $newName")
+            } else {
+                log.info("Component with name $name will be updated to $newName")
+            }
+        } ?: run {
+            log.warn("Component with name $name not found in DMS")
+            if (component == null) {
+                throw NotFoundException("Component with name $name not found in DMS")
```

Comment:
None of $name and $newName components were found in DMS

Reply:
fixed